Resources/config/config.xml

```diff
@@ -58,8 +58,8 @@
             <tag name="twig.extension" alias="fos_message" />
         </service>
 
-        <service id="fos_message.search_query_factory.default" class="FOS\MessageBundle\Search\QueryFactory" scope="request" public="true">
-            <argument type="service" id="request" />
+        <service id="fos_message.search_query_factory.default" class="FOS\MessageBundle\Search\QueryFactory" public="true">
+            <argument type="service" id="request_stack" />
```

Comment:
this should be managed in a compiler pass or we must remove support of ``symfony/framework-bundle < 2.4``.